pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -780,6 +783,46 @@ func lookupDomainConfig(ctx *domainContext, key string) *types.DomainConfig {
 	return &config
 }
 
+func execWithTimeout(command string, args ...string) ([]byte, bool, error) {
+
+	ctx, cancel := context.WithTimeout(context.Background(),
+		10*time.Second)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, command, args...)
+	out, err := cmd.Output()
+	if ctx.Err() == context.DeadlineExceeded {
+		return nil, false, nil
+	}
+	return out, true, err
+}
+
+// Execute rkt commands
+func ExecuteRktCmd(URL string) []byte {
+
+	cmd := "rkt"
+	arg1 := "run"
+	//arg2 := "registry-1.docker.io/zededa/zcli:latest"
+	arg2 := URL
+	//arg3 := "--interactive"
+	arg3 := "--insecure-options=image"
+	arg4 := "--stage1-path=/usr/sbin/stage1-xen.aci"
+
```

Comment:
@spatchineelam I see you're constructing the command like for rkt run -- we will have to pass an extra argument of xl.cfg constructed by domainmanager to rkt as well. This doesn't have to be part of your first cut, but we need to structure the code in such a way that would allow us to do that very quickly. See my comments below on how to restructure the code.

Reply:
When I was going thru the domain manager code yesterday, I had the same question about configurability of stage1-xen. I will rearrange the code